docs/Passing-Devices-with-Clear-Containers.md

```diff
@@ -0,0 +1,68 @@
+# Using devices with Clear Containers
+
+You can use the docker run command with the --device option to pass host 
+devices to a container. The following example passes a device:
+```
+$ sudo docker run -it --device=/dev/sdb debian
+```
+Clear Containers supports passing the following devices to the
+container with `--device`.
+
+## Virtual Function I/O (VFIO) Devices (PCI Passthrough)
+
+The VFIO driver is a framework for exposing direct device access to userspace, 
+in a secure, Input–Output Memory Management Unit (IOMMU) protected environment.
+A device can be directly passed to a Clear Containers container using VFIO. 
+
+For information on VFIO, refer to the following links:
+ - https://www.kernel.org/doc/Documentation/vfio.txt
+ - https://www.linux-kvm.org/images/b/b4/2012-forum-VFIO.pdf
+
+Devices can pe passed to a Clear Container container using VFIO
+as shown below:
+
+```
+$ sudo docker run -it --device=/dev/vfio/16 centos/tools bash
+```
+You are required to unbind the device from its host driver, bind the device
+to `vfio-pci` driver and then pass the IOMMU group that the device belongs to on
+the docker command line.
+For detailed steps to bind and unbind devices, see https://github.com/containers/virtcontainers#how-to-pass-a-device-using-vfio-passthrough
+
+
+## Block Devices
+
+Clear Containers passes block devices to containers using `virtio-block` when you
+use --device on the docker command line to pass block devices.
+The file system that is present on the block device should be enabled in the 
+Clear Container kernel. We currently have support for ext4, xfs and overlay file
+systems in the Clear Container kernel.
+
+An example of passing a block device using a fake image is shown below:
```

Comment:
Suggested rewrite for consistency with the following example ... device rename support for block devices: The following example passes a block device using a fake image:

Reply:
@klynnrif Fixed this.

Reply:
+1